src/Form/AuthenticationForm.php

```diff
@@ -456,77 +490,85 @@ protected function createSuggestion(\Exception $exception, KeyInterface $key): M
-      $credentials['username'] = $key_type->getUsername($key);
-    }
-    catch (AuthenticationKeyValueMalformedException $exception) {
-      // Could not read the credentials because the key value storage is
-      // malformed.
-    }
+    $credentials = !($key_type instanceof EdgeKeyTypeInterface) ? [] : [
+      'endpoint' => $key_type->getEndpoint($key),
+      'organization' => $key_type->getOrganization($key),
+      'username' => $key_type->getUsername($key),
+    ];
 
     $keys = [
-      'key_type' => get_class($key_type),
+      'auth_type' => ($key_type instanceof EdgeKeyTypeInterface) ? $key_type->getAuthenticationType($key) : 'invalid credentials',
       'key_provider' => get_class($key->getKeyProvider()),
     ];
 
-    // TODO
-    // * Remove deprecated code.
-    // * Fix implementation of this method.
-    // * Provide debug message for OauthTokenStorageException.
-    if ($key_type instanceof OauthKeyType) {
-      /** @var \Drupal\apigee_edge\Plugin\KeyType\OauthKeyType $key_type */
+    if (!empty($credentials) && $keys['auth_type'] === EdgeKeyTypeInterface::EDGE_AUTH_TYPE_OAUTH) {
       $credentials['authorization_server'] = $key_type->getAuthorizationServer($key);
       $credentials['client_id'] = $key_type->getClientId($key);
       $credentials['client_secret'] = $key_type->getClientSecret($key) === Oauth::DEFAULT_CLIENT_SECRET ? Oauth::DEFAULT_CLIENT_SECRET : '***client-secret***';
-
-      $keys['key_token_type'] = get_class($key_token->getKeyType());
-      $keys['key_token_provider'] = get_class($key_token->getKeyProvider());
     }
 
-    $exception_text = (string) $exception;
-    $exception_text = preg_replace('/(.*refresh_token=)([^\&\r\n]+)(.*)/', '$1***refresh-token***$3', $exception_text);
-    $exception_text = preg_replace('/(.*mfa_token=)([^\&\r\n]+)(.*)/', '$1***mfa-token***$3', $exception_text);
-    $exception_text = preg_replace('/(.*password=)([^\&\r\n]+)(.*)/', '$1***password***$3', $exception_text);
-    $exception_text = preg_replace('/(Authorization: (Basic|Bearer) ).*/', '$1***credentials***', $exception_text);
-
-    $text = json_encode($credentials, JSON_PRETTY_PRINT) .
-      PHP_EOL .
-      json_encode($keys, JSON_PRETTY_PRINT) .
-      PHP_EOL .
-      json_encode($this->config('apigee_edge.client')->get(), JSON_PRETTY_PRINT) .
-      PHP_EOL .
+    // Sanitize exception text.
+    $exception_text = preg_replace([
+      '/(.*refresh_token=)([^\&\r\n]+)(.*)/',
+      '/(.*mfa_token=)([^\&\r\n]+)(.*)/',
+      '/(.*password=)([^\&\r\n]+)(.*)/',
+      '/(Authorization: (Basic|Bearer) ).*/',
+    ], [
+      '$1***refresh-token***$3',
+      '$1***mfa-token***$3',
+      '$1***password***$3',
+      '$1***credentials***',
+    ], (string) $exception);
+
+    return json_encode($credentials, JSON_PRETTY_PRINT) . PHP_EOL .
+      json_encode($keys, JSON_PRETTY_PRINT) . PHP_EOL .
+      json_encode($this->config('apigee_edge.client')->get(), JSON_PRETTY_PRINT) . PHP_EOL .
```

Comment:
It prints the config hash too:
`{
   “http_client_connect_timeout”: 30,
   “http_client_timeout”: 30,
   “http_client_proxy”: “”,
   “_core”: {
       “default_config_hash”: “k8IHdAQdw4tE74jPYyd8LGjY0mbjOBGrj5bBzhGISf8”
   }
}`

Reply:
This is not a major one, but it would be nice if the `_core` key and its value would not be printed.

Reply:
I filtered out anything beginning with an `_`.